segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/SegmentMetadataUpdateTransaction.java

```diff
@@ -397,7 +397,7 @@ void preProcessOperation(DeleteSegmentOperation operation) {
      * @throws IllegalArgumentException        If the operation is for a different Segment.
      */
     void preProcessAsTargetSegment(MergeSegmentOperation operation, SegmentMetadataUpdateTransaction sourceMetadata)
-            throws StreamSegmentSealedException, StreamSegmentNotSealedException, MetadataUpdateException {
+            throws StreamSegmentSealedException, StreamSegmentNotSealedException, MetadataUpdateException, BadAttributeUpdateException {
```

Comment:
Update Javadoc too to reflect the new exception that might be thrown.

Reply:
Done.